### PR TITLE
Desktop: Fixes #9960: Creating a profile changes the language of Joplin

### LINKED ISSUE #9960: Creating a profile changes the language of Joplin
### Operating system

Windows

### Joplin version

2.14.14

### Desktop version info

Joplin 2.14.14 (prod, win32)

Client ID: 0e33609c84614b3092102ff00887b36e
Sync Version: 3
Profile Version: 45
Keychain Supported: Yes

Revision: ecb6ac3

### Current behaviour

1. Change your Joplin language other than the operating system language
2. Create a new profile
3. The language for the new profile is the same as for the operating system 
4. Change back to your original profile 
5. The language has now been changed to that of the operating system here too

### Expected behaviour

The language should not be changed automatically when creating a new Joplin profile

### Logs

_No response_

### PR DESCRIPTION
**Pull request by cagnusmarlsen**
### Summary
This fixes the issue where creating a new profile changes the languages of all profiles including the default profile to the system default language.

Fixes #9960.

### Testing
Follow reproduction steps in #9960:

1. Change your Joplin language other than the operating system language
2. Create a new profile
3. The language for the new profile is the same as for the original profile
4. Change back to your original profile
5. Verify that the language of the original profile has not changed

This has been tested successfully on Windows 11 and Ubuntu.

### CODE REVIEW

**Comment by cagnusmarlsen** [comment]
@PackElend label me please

File: packages/lib/BaseApplication.ts [hunk 1]
```diff
@@ -747,8 +747,10 @@ export default class BaseApplication {
 		}
 
 		if (Setting.value('firstStart')) {
-			const locale = shim.detectAndSetLocale(Setting);
-			reg.logger().info(`First start: detected locale as ${locale}`);
+			if (!Setting.value('isSubProfile')) {
```
Comment (laurent22) [inline packages/lib/BaseApplication.ts:755]: Thank for this fix. Would you mind adding a short comment to explain why we have this check?

Comment (cagnusmarlsen) [reply packages/lib/BaseApplication.ts:755]: Not at all, I've added a comment to the code.

**Comment by cagnusmarlsen** [comment]
### PR Summary:

- **Changes**: This PR introduces a new feature to the Joplin application. The objective is to enhance the user interface with a new settings panel and improve the user experience.

- **PR Type**: Feature

- **Important Change Files**:
  - `src/gui/SettingsPanel.tsx`: This file contains the main logic for the new settings panel.
  - `src/gui/Header.tsx`: Adjustments to integrate the new settings panel with the existing header.
  - `src/styles/settingsPanel.css`: Styling for the new settings panel to match the application's overall look and feel.

**Pull request merged**